pkg/kubelet/util/pluginwatcher/plugin_watcher.go

```diff
@@ -153,25 +149,7 @@ func (w *Watcher) Start() error {
 // Stop stops probing the creation of plugin sockets at the path
 func (w *Watcher) Stop() error {
```

Comment:
Since the previous PR https://github.com/kubernetes/kubernetes/pull/75110 merged, `Stop` func would called only when `traverse` error occurs when the next progress would not go on. So waitGroup seems to be useless and we do not extra time to wait.

Reply:
It sounds like the `waitGroup` is not useful with the current way the `Stop` func is used. However, can we be guaranteed that the `Stop` func won't be used in a different way in the future, and that, when used in that different way, we would still need the `waitGroup`?

Basically, I'm wondering if we're changing the `Stop` function to be more fragile based on implementation details which aren't guaranteed to be true in the future.